homeassistant/components/homekit/thermostats.py

```diff
@@ -0,0 +1,135 @@
+from .accessories import HomeAccessory
+from .const import (SERV_THERMOSTAT, CHAR_CURRENT_HEATING_COOLING,
+                    CHAR_TARGET_HEATING_COOLING, CHAR_CURRENT_TEMPERATURE,
+                    CHAR_TARGET_TEMPERATURE, CHAR_TEMP_DISPLAY_UNITS)
+
+_LOGGER = logging.getLogger(__name__)
+
+UNIT_HASS_TO_HOMEKIT = {'°C': 0, '°F': 1}
+UNIT_HOMEKIT_TO_HASS = {c: s for s, c in UNIT_HASS_TO_HOMEKIT.items()}
+HC_HASS_TO_HOMEKIT = {'off': 0, STATE_HEAT: 1, STATE_COOL: 2}
+HC_HOMEKIT_TO_HASS = {c: s for s, c in HC_HASS_TO_HOMEKIT.items()}
+
+
+@TYPES.register('Thermostat')
+class Thermostat(HomeAccessory):
+    """Generate a Thermostat accessory for an alarm control panel."""
+
+    def __init__(self, hass, entity_id, display_name):
+        """Initialize a Thermostat accessory object."""
+        super().__init__(display_name, entity_id, 'THERMOSTAT')
+
+        self._hass = hass
+        self._entity_id = entity_id
+        self._call_timer = None
+
+        self.current_heat_cool = None
+        self.homekit_target_heat_cool = None
+        self.current_temperature = None
+        self.homekit_target_temperature = None
+        self.target_temperature = None
+        self.display_units = None
+
+        self.service_thermostat = self.get_service(SERV_THERMOSTAT)
+        self.char_current_heat_cool = self.service_thermostat.\
+            get_characteristic(CHAR_CURRENT_HEATING_COOLING)
+        self.char_target_heat_cool = self.service_thermostat.\
+            get_characteristic(CHAR_TARGET_HEATING_COOLING)
+        self.char_target_heat_cool.setter_callback = self.set_heat_cool
+
+        self.char_current_temp = self.service_thermostat.\
+            get_characteristic(CHAR_CURRENT_TEMPERATURE)
+        self.char_target_temp = self.service_thermostat.\
+            get_characteristic(CHAR_TARGET_TEMPERATURE)
+        self.char_target_temp.setter_callback = self.set_temperature_debounced
+
+        self.char_display_units = self.service_thermostat.\
+            get_characteristic(CHAR_TEMP_DISPLAY_UNITS)
+
+    def run(self):
+        """Method called be object after driver is started."""
+        state = self._hass.states.get(self._entity_id)
+        self.update_thermostat(new_state=state)
+
+        async_track_state_change(self._hass, self._entity_id,
+                                 self.update_thermostat)
+
+    def set_heat_cool(self, value):
+        """Move operation mode to value if call came from homekit."""
+        if value != self.current_heat_cool and value in HC_HOMEKIT_TO_HASS:
+            _LOGGER.debug("%s: Set heat-cool to %d", self._entity_id, value)
+            self.homekit_target_heat_cool = value
+            hass_value = HC_HOMEKIT_TO_HASS[value]
+
+            self._hass.services.call('climate', 'set_operation_mode',
+                                     {'entity_id': self._entity_id,
+                                      'operation_mode': hass_value})
+
+    def set_temperature_debounced(self, value):
```

Comment:
Is this necessary?

Reply:
Necessary is a big word off course :). But I noticed for myself that if you use the Home app to set the temperature, sliding the temperature from 21 to 16 degrees, causes set temperature calls to 20, 19, 18, 17 and 16, while all I want to do is set the temperature to 16 degrees. To avoid sending to much set calls to the device (Nest thermostat in my case), I though debouncing was a good idea.

Reply:
I get the idea, but you can select a discrete value in HomeKit. Which only results in one call to the method anyway.